phpcs.base.xml

```diff
@@ -0,0 +1,184 @@
+        <exclude-pattern>*/widgets/*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/layouts/*\.php</exclude-pattern>
+        <exclude-pattern>*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/views/*\.php</exclude-pattern>
+    </rule>
+
+    <rule ref="PSR12.ControlStructures.ControlStructureSpacing.CloseParenthesisIndent">
+        <!--
+        Partials can be a little more lenient with closing parenthesis indentation if it looks better within HTML templating code
+        -->
+        <exclude-pattern>*/behaviors/*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/components/*/*\.php</exclude-pattern>
+        <exclude-pattern>*/controllers/*/*\.php</exclude-pattern>
+        <exclude-pattern>*/formwidgets/*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/reportwidgets/*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/widgets/*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/layouts/*\.php</exclude-pattern>
+        <exclude-pattern>*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/views/*\.php</exclude-pattern>
+    </rule>
+
+    <rule ref="Generic.Arrays.ArrayIndent">
+        <!--
+        Partials may format arrays differently to improve readability within HTML code
+        -->
+        <exclude-pattern>*/behaviors/*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/components/*/*\.php</exclude-pattern>
+        <exclude-pattern>*/controllers/*/*\.php</exclude-pattern>
+        <exclude-pattern>*/formwidgets/*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/reportwidgets/*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/widgets/*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/layouts/*\.php</exclude-pattern>
+        <exclude-pattern>*/partials/*\.php</exclude-pattern>
+        <exclude-pattern>*/views/*\.php</exclude-pattern>
+    </rule>
+
+    <!-- All arrays must have a trailing comma on the last item in the array -->
+    <rule ref="SlevomatCodingStandard.Arrays.TrailingArrayComma" />
+    <!-- Imports should be alphabetically ordered -->
+    <rule ref="SlevomatCodingStandard.Namespaces.AlphabeticallySortedUses" />
+    <!-- Don't allow group use declarations as they make diffs ugly -->
+    <rule ref="SlevomatCodingStandard.Namespaces.DisallowGroupUse" />
+    <!-- Don't allow use definitions for a class in the same namespace -->
+    <rule ref="SlevomatCodingStandard.Namespaces.UseFromSameNamespace" />
+    <!-- Don't allow starting backslash for imports as there's no need for them -->
+    <rule ref="SlevomatCodingStandard.Namespaces.UseDoesNotStartWithBackslash" />
+    <!-- Don't allow unused imports (using them in docblocks is OK)-->
+    <rule ref="SlevomatCodingStandard.Namespaces.UnusedUses">
+        <properties>
+            <property name="searchAnnotations" value="true" />
+        </properties>
+    </rule>
+</ruleset>
```

Comment:
This file should probably live in the system module and be referenced from the stubs as such to avoid having the core ruleset changed on a per project basis and also to support existing projects making use of this feature when they don't have this file present since it's not distributed through a composer module update.